### ISSUE TITLE
Watch `.git/info/exclude` for changes and re-run `git status`

### ISSUE THREAD

**Issue by mathieucaroff** [microsoft/vscode#98945]
- VSCode Version:
  - Version: 1.46.0-insider (user setup)
  - Commit: cd41dd1f6065db76c77c6e2f659bb50e6f558dec
  - Date: 2020-05-29T05:51:32.662Z
  - Electron: 7.3.0
  - Chrome: 78.0.3904.130
  - Node.js: 12.8.1
  - V8: 7.8.279.23-electron.0
  - OS: Windows_NT x64 10.0.18363
- OS Version: Windows 10 v1909

Does this issue occur when all extensions are disabled?: **Yes, it does**

Steps to Reproduce:

0. (Make sure the VSCode's Explorer panel is open (Ctrl+Shift+E) -- this is where we will notice the issue.)
1. Create a new git repository with a `hello.txt` file -- `git init; touch hello.txt`
2. Open the local git exclude file of the local repository: `.git/info/exclude`
3. Add `hello.txt` to that exclude file, and save it.

Expected behavior:

hello.txt is grey out in the Explorer as soon as the `exclude` file is saved.

Actual behavior:

hello.txt stays green (i.e. reported as untracked file)

Note:

- (A) This issue can also be noticed from the Source Control Panel (Ctrl+Shift+G)
- (B) Opening and saving .gitignore makes VSCode refresh the git status, removing the glitched state
- (C) Running `git status` in the console partially removes the glitched state:
  - Files that have been added to `.git/info/exclude` will no longer appear untracked, but they won't appear ignored either
  - Files that have been removed from `.git/info/exclude` will appear untracked, as expected.

**Comment by tabaddor** [comment]
I can look into this issue. What have you been thinking @mathieucaroff ?

**Comment by mathieucaroff** [comment]
I'm not familiar with the source code of VSCode, so I don't know how the code is structured.

Hopefully, the way git is managed does not involve evolving an internal copy of a the state of the repository, i.e. hopefully, VSCode uses a single-source of truth (git) approach to this problem, and all we have to do is force a refresh.

If yes, I expect the following approach to be easy:

- Figure out in VSCode, what service currently manages watching the .gitignore file
  - If it only support watching a single file, extend or change it, to let it support several
- Configure the service to watch not only `.gitignore`, but also `.git/info/exclude`.

**Comment by Gum-Joe** [comment]
I'm happy to explore a fix here.  I had a look at the VSCode git extension yesterday briefly and managed to track down the repo refresh & watch functions, but nothing to do with .gitignore watching yet.  I'll have a better look if I start work on this, and will be able to clone & explore the repo.

**Comment by mathieucaroff** [comment]
I cloned the repository and searched for `.gitignore` (I got 42 results).

I believe I found where the `.gitignore` watching is set up: in [`decorationProvider.ts`, in the constructor of `GitIgnoreDecorationProvider`](https://github.com/microsoft/vscode/blob/af9d0d5df2679d6dd0c2fd9f99d69e4c79376f8b/extensions/git/src/decorationProvider.ts#L24).

The chain then goes as follow:

- `GitIgnoreDecorationProvider` is used in `GitDecorations.constructor` (same file).
- `GitDecorations` is used in `createModel()`, in the main.ts of the git extension.

I'm tempted to think that modifying the constructor of `GitIgnoreDecorationProvider` should be sufficient.
